src/DotNetWorker.ApplicationInsights/FunctionsTelemetryModule.cs

```diff
@@ -20,18 +27,37 @@ public void Initialize(TelemetryConfiguration configuration)
 
             _listener = new ActivityListener
             {
-                ShouldListenTo = source => source.Name.StartsWith("Microsoft.Azure.Functions.Worker"),
+                ShouldListenTo = source => source.Name.StartsWith(TraceConstants.FunctionsActivitySource),
                 ActivityStarted = activity =>
                 {
                     var dependency = _telemetryClient.StartOperation<DependencyTelemetry>(activity);
```

Comment:
I was reading through `StartOperation` and I noticed it will call [Activity.Start](https://github.com/microsoft/ApplicationInsights-dotnet/blob/main/BASE/src/Microsoft.ApplicationInsights/TelemetryClientExtensions.cs#L330) - but this is an already started activity. This will cause an [exception to be raised](https://github.com/dotnet/runtime/blob/main/src/libraries/System.Diagnostics.DiagnosticSource/src/System/Diagnostics/Activity.cs#L683) and immediately caught as this activity will always be already started.

This might not be a big deal, just something to think about.

Reply:
Yes, I noticed as well. The problem is there's so much subtle code inside of the operation tracking w/ Activity that I didn't see a great way around it (I didn't want to copy all that logic). Can follow up with App Insights separately before we GA that to see if there's a better way.